pkgs/by-name/xd/xdg-desktop-portal-termfilechooser/package.nix

```
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, meson
+, ninja
+, pkg-config
+, wayland-protocols
+, wayland-scanner
+, inih
+, libdrm
+, mesa
+, scdoc
+, systemd
+, wayland
+}:
+
+stdenv.mkDerivation {
+  pname = "xdg-desktop-portal-termfilechooser";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "nani8ot";
+    repo = "xdg-desktop-portal-termfilechooser";
+    rev = "4e0db3d4c1639582420847393884ca6bb990cfe5";
+    hash = "sha256-4s3n5EONwmDhdMrs7NdCqM/IAcNXmmbuhv84btCredg=";
+  };
+
+  strictDeps = true;
+
+  depsBuildBuild = [ pkg-config ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    scdoc
+    wayland-scanner
+    makeWrapper
+  ];
+
+  buildInputs = [
+    inih
+    libdrm
+    mesa
+    systemd
+    wayland
+    wayland-protocols
+  ];
+
+  mesonFlags = [
+    (lib.mesonOption "sd-bus-provider" "libsystemd")
+  ];
+
```

Since the ranger wrapper is installed and [enabled out of the box](https://github.com/GermainZ/xdg-desktop-portal-termfilechooser/blob/71dc7ab06751e51de392b9a7af2b50018e40e062/src/core/config.c#L11), maybe fix the path:

```suggestion

  postPatch = ''
    # Allow using ranger out of the box without any configuration.
    substituteInPlace src/core/config.c \
      --replace-fail '"/usr/share/xdg-desktop-portal-termfilechooser/ranger-wrapper.sh"' '"${placeholder "out"}/share/xdg-desktop-portal-termfilechooser/ranger-wrapper.sh"'
  '';
```

---

Nix tells me, when building, that `--replace-fail` is not a valid option?

---

Might need to rebase onto newer revision of Nixpkgs. It was only introduced recently: https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh#L834